### PR TITLE
[Inference] use fp8 cuda core gemm kernel when M<=4

### PR DESCRIPTION
**Pull request by zhink**
### PR types

New features 
### PR changes

Others
### Description

To speed up fp8 gemm calculations when m<=4, you should export FLAGS_cuda_core_fp8_gemm=1 to use it.
speed data is :
<meta charset="UTF-8"><meta charset="UTF-8"><div class="mp-table-align"><div class="mp-table-wrapper"><div data-slate-node="element" style="padding-left:0px" class="mp-table-container">
m | k | n | L20(秒) | L20(秒) | 加速(%)
-- | -- | -- | -- | -- | --
1 | 4096 | 4096 | 5.43 | 3.85 | 29.05
1 | 4096 | 12800 | 5.45 | 3.86 | 29.21
1 | 6144 | 4096 | 5.43 | 3.89 | 28.34
1 | 2048 | 2048 | 5.43 | 3.91 | 28.13
1 | 2048 | 5504 | 5.55 | 3.93 | 29.30
1 | 6144 | 2048 | 5.45 | 3.90 | 28.43
1 | 5120 | 5120 | 5.45 | 3.90 | 28.56
1 | 5120 | 13824 | 5.50 | 3.97 | 27.74
1 | 15360 | 5120 | 5.56 | 3.93 | 29.28
2 | 4096 | 4096 | 5.51 | 3.97 | 27.97
2 | 4096 | 12800 | 5.52 | 3.94 | 28.54
2 | 6144 | 4096 | 5.50 | 3.95 | 28.13
2 | 2048 | 2048 | 5.51 | 3.93 | 28.57
2 | 2048 | 5504 | 5.58 | 3.95 | 29.20
2 | 6144 | 2048 | 5.49 | 3.95 | 28.13
2 | 5120 | 5120 | 5.50 | 3.95 | 28.19
2 | 5120 | 13824 | 5.49 | 3.94 | 28.26
2 | 15360 | 5120 | 5.59 | 4.04 | 27.62
3 | 4096 | 4096 | 5.55 | 3.96 | 28.59
3 | 4096 | 12800 | 5.56 | 3.96 | 28.73
3 | 6144 | 4096 | 5.47 | 3.93 | 28.15
3 | 2048 | 2048 | 5.49 | 3.95 | 28.19
3 | 2048 | 5504 | 5.58 | 3.95 | 29.22
3 | 6144 | 2048 | 5.53 | 3.93 | 28.84
3 | 5120 | 5120 | 5.46 | 3.93 | 28.04
3 | 5120 | 13824 | 5.43 | 4.31 | 20.58
3 | 15360 | 5120 | 5.53 | 5.25 | 4.99
4 | 4096 | 4096 | 5.48 | 3.89 | 29.05
4 | 4096 | 12800 | 5.49 | 4.03 | 26.65
4 | 6144 | 4096 | 5.47 | 3.89 | 28.74
4 | 2048 | 2048 | 5.46 | 3.89 | 28.69
4 | 2048 | 5504 | 5.53 | 3.90 | 29.40
4 | 6144 | 2048 | 5.47 | 3.91 | 28.45
4 | 5120 | 5120 | 5.48 | 3.91 | 28.67
4 | 5120 | 13824 | 5.45 | 5.22 | 4.12
4 | 15360 | 5120 | 5.64 | 6.38 | -13.14

</div><div class="mp-table-serialize-flag"><br/></div></div></div><span data-morpho-doc-data='{"token":"eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIiwiYXBwSWQiOjEsInVpZCI6IkcwclhMb2E0dXMiLCJkb2NJZCI6IkE2QjJONF9Gb3F0MnpuIn0..nHZgx_FuOt_RBHf6.jqDuFtm_uJDlQvOUhb2EVKhFQdqnFBm0jxK2O0ZjF_NDSpBw2idmC0RqfLbaScpzAYY1upjR_r3KxTWpJFhtAvtZbdfOJIBhDbF3UzZ1TaqwUBm1afJlTeSlSjdon1q3JodGRqVtfjru8w3W8AaJtm0s2w3Q7i19vJFjYmq9rkroDrDcALZ25E2DO0yC5XoHmudIX6fRlqGmhptQjPzcEuAtQQ.GCiYEgk5N6QNuaWq3bGZPA","appId":"1"}' class='mp-morpho-clipboard-doc-data'>﻿</span>

### CODE REVIEW

File: llm/docs/predict/best_practices.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ PaddleNLP 提供了多种环境变量，用于优化推理性能和资源使用
 
 - `FLAGS_cublaslt_device_best_config`：在 FLAGS_enable_blaslt_global_search 设为1的前提下，使用`FLAGS_cublaslt_device_best_config`来指定离线调优出的 int8 gemm 配置文件，默认值为""。配置文件可以通过`PaddleNLP/csrc/utils/tune_cublaslt_int8_gemm.py`产出，该脚本会自动搜索当前输入大小下 cuBLASLt 提供的最优 gemm 配置并将结果记录下来，需要注意的是不同的 CUDA 版本需要分别 tune。推理 A8W8模型并且 FLAGS_enable_blaslt_global_search 设为1时使用此 FLAG 会获得更优的性能。
 
+- `FLAGS_cuda_core_int8_gemm`：是否开启小 Batch Int8 Gemm优化，默认值不开启。设为1可开启，推理A8W8模型时，性能会更好。
```
Comment (DrownFish19) [inline llm/docs/predict/best_practices.md]: 这里的性能会更好比较虚，现在已有数据可以贴上

File: llm/docs/predict/best_practices.md [hunk 2]
```diff
@@ -9,6 +9,10 @@ PaddleNLP 提供了多种环境变量，用于优化推理性能和资源使用
...
+
+- `FLAGS_cuda_core_fp8_gemm`：是否开启小 Batch FP8 Gemm优化，默认值不开启。设为1可开启，推理 FP8模型时，性能会更好。
```
Comment (DrownFish19) [inline llm/docs/predict/best_practices.md]: “性能会更好”，修改建议同上

**Comment by DrownFish19** [comment]
1. 推荐把这个加速优化写到文档里；
2. 在Flag说明里加上加速比和限制条件。

File: llm/docs/predict/best_practices.md (hunk 2 above)
Comment (zhink) [reply llm/docs/predict/best_practices.md]: ok

**Review by DrownFish19 — APPROVED** [review]

File: llm/docs/predict/best_practices.md (hunk 1 above)
Comment (zhink) [reply llm/docs/predict/best_practices.md]: ok

**Pull request merged**